### ISSUE TITLE
[Docker-TypeScript] Using ts-node and nodemon doesn't properly

### ISSUE THREAD

**Issue by umarov** [Microsoft/vscode-recipes#187]
I have an [example project here](https://github.com/umarov/note-taker).

My nodemon command with ts-node looks like this inside `nodemon-debug.json`:

```shell
node --inspect=0.0.0.0:9229 --nolazy -r ts-node/register -r tsconfig-paths/register src/main.ts
```
The debug command that gets executed by docker is this:

```shell
nodemon --config nodemon-debug.json
```

Dockerfile looks like this:
```dockerfile
FROM node:10-alpine

RUN mkdir -p /usr/src/app

WORKDIR /usr/src/app

COPY ["package.json", "yarn.lock", "npm-shrinkwrap.json*", "./"]

RUN yarn install --silent && mv node_modules ../

COPY . .

EXPOSE 9229
EXPOSE 3000

RUN chmod u+x ./scripts/start.sh

CMD ./scripts/start.sh
```

launch.json looks like this:

```json
{
  "type": "node",
  "request": "attach",
  "name": "Attach to Docker",
  "port": 9229,
  "address": "0.0.0.0",
  "restart": true,
  "localRoot": "${workspaceFolder}",
  "remoteRoot": "/usr/src/app",
  "skipFiles": [
    "<node_internals>/**/*.js",
  ]
}
```

VS Code attaches, but when I put a breakpoint, it looks like this:
![image](https://user-images.githubusercontent.com/2321915/51722075-31eb6c80-2022-11e9-883b-662db0ee2d5b.png)
![image](https://user-images.githubusercontent.com/2321915/51722156-77a83500-2022-11e9-982a-534d7c515b97.png)

I tried to follow the TypeScript Docker guide as closely as possible. My setup is a bit different. I was wondering if it would be possible to add an example that uses *nodemon* and *ts-node* the way I am using it. Or tips on getting it to work. I could make a PR to add directions if I can get this to work. 

The nodemon script I use it generated from Nest.js. I change the inspect from `--inspect-brk` to `--inspect=0.0.0.0:9229`.

**Comment by umarov** [comment]
Setting breakpoints and stepping through code works fine in Google Chrome. 

![image](https://user-images.githubusercontent.com/2321915/51722601-e934b300-2023-11e9-8526-84f18ec33d2e.png)

I pick the `sm` (source mapped) version and it works like expected. It doesn't work the same way in VS Code. When I have both Chrome and VS Code connected at the same time. VS Code gets triggered by the same breakpoint I set in Chrome. But the file that's shown is the read-only source map version. Breakpoints work fine on that version in VS Code too, but on the original `.ts` file the breakpoints don't stick.

**Comment by cschroeter** [comment]
I have the same issue. Not sure what to do

**Comment by eddiemonge** [comment]
I have the same issue. You can also take nodemon out of the equation to simplify

**Comment by eddiemonge** [comment]
> You can use ts-node if your local root and the remote root are the same.

Yes this. Matching the local and remote roots and changing the workdir in docker fixed it but that is not a good fix. not portable at all.

**Comment by eddiemonge** [comment]
The `sourceMapPathOverrides` option fixes it for me:

`launch.json`
```
    {
      "type": "node",
      "request": "attach",
      "name": "Docker: Attach to Node",
      "sourceMapPathOverrides": {
        "/usr/app/*": "${workspaceRoot}/*"
      }
    },
```

`Dockerfile`
```
WORKDIR /usr/app
```

**Comment by Kristof-Mattei** [comment]
This is still an issue.

While the debugger can attach, the location on which the breakpoints are set is wrong:

![image](https://user-images.githubusercontent.com/864376/82112238-f1749e00-96ff-11ea-809b-b6bdf55ec87b.png)

This after setting the right kind of `sourceMapPathOverrides`.

**Comment by connor4312** [comment]
We no longer have a Docker-TypeScript recipe as it has been supplanted by [Remote-Containers](https://code.visualstudio.com/docs/remote/containers) functionality.
